bgpd/bgp_io.c

```diff
@@ -466,7 +466,8 @@ static uint16_t bgp_read(struct peer *peer, int *code_p)
 	uint16_t status = 0;
 
 	readsize =
-		MIN(ringbuf_space(peer->ibuf_work), sizeof(peer->ibuf_scratch));
+		MIN(ringbuf_space(peer->ibuf_work),
+		    (BGP_EXTENDED_MESSAGE_MAX_PACKET_SIZE * peer->rpkt_quanta));
```

Comment:
Instead of this, store the size of this buffer right next to the buffer. Copy-pasting the definition of the size is a recipe for a buffer overflow.

Reply:
ok